packages/checkout-extension/src/Extension.tsx

```diff
@@ -0,0 +1,42 @@
+import { ExtensionRegion } from '@bigcommerce/checkout-sdk';
+import React, { useEffect } from 'react';
+
+import { useCheckout } from '@bigcommerce/checkout/payment-integration-api';
+
+import { extensionRegionToContainerMap } from './ExtensionRegionContainer';
+import { useExtensions } from './useExtensions';
+
+interface ExtensionProps {
+    region: ExtensionRegion;
+}
+
+export const Extension = ({ region }: ExtensionProps): React.JSX.Element | null => {
+    const {
+        checkoutState: {
+            data: { getExtensions, getConfig, getCheckout },
+        },
+    } = useCheckout();
+    const extensions = getExtensions();
+    const config = getConfig();
+    const checkout = getCheckout();
+    const { extensionService } = useExtensions();
+    const isRegionEnabled =
+        extensions && config && checkout && extensionService.isRegionEnabled(region);
+    const containerId = extensionRegionToContainerMap[region];
+
+    useEffect(() => {
+        if (isRegionEnabled) {
+            void extensionService.renderExtension(containerId, region);
+
+            return () => {
+                extensionService.removeListeners(region);
+            };
+        }
+    }, [containerId, extensionService, isRegionEnabled, region]);
```

Comment:
To avoid placing `useEffect` across the repo for each region, I think it makes sense to build a component that handles everything, including creating the container for a region.